scraper.go

```diff
@@ -28,11 +28,12 @@ type ScraperTarget struct {
 
 // ScraperTargetStoreService defines the crud service for ScraperTarget.
 type ScraperTargetStoreService interface {
+	UserResourceMappingService
 	ListTargets(ctx context.Context) ([]ScraperTarget, error)
-	AddTarget(ctx context.Context, t *ScraperTarget) error
+	AddTarget(ctx context.Context, t *ScraperTarget, userID ID) error
```

Comment:
Why does this function take a userID?

Reply:
this is for creating the user resource mapping.

Reply:
I see. I think for now this is fine, but we should probably revisit this pattern to see if this is something we want to repeat.

Reply:
Specifically, I'm concerned about us mixing abstractions and tightly coupling pieces together.

Reply:
👍 yeah, totally agree. I think it's better maybe we can think about adapting a search engine thing for the indexing.